deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -325,60 +231,181 @@ def hash_bucket_input_provider(index, item):
         **params.deltacat_storage_kwargs,
     )
 
-    # BSP Step 2: Merge
-    merge_options_provider = functools.partial(
+    hb_options_provider = functools.partial(
         task_resource_options_provider,
         pg_config=params.pg_config,
-        resource_amount_provider=merge_resource_options_provider,
-        num_hash_groups=params.hash_group_count,
-        hash_group_size_bytes=all_hash_group_idx_to_size_bytes,
-        hash_group_num_rows=all_hash_group_idx_to_num_rows,
-        round_completion_info=round_completion_info,
-        compacted_delta_manifest=previous_compacted_delta_manifest,
+        resource_amount_provider=hash_bucket_resource_options_provider,
+        previous_inflation=params.previous_inflation,
+        average_record_size_bytes=params.average_record_size_bytes,
         primary_keys=params.primary_keys,
-        deltacat_storage=params.deltacat_storage,
-        deltacat_storage_kwargs=params.deltacat_storage_kwargs,
         ray_custom_resources=params.ray_custom_resources,
     )
 
-    def merge_input_provider(index, item):
-        return {
-            "input": MergeInput.of(
-                dfe_groups_refs=item[1],
-                write_to_partition=compacted_partition,
-                compacted_file_content_type=params.compacted_file_content_type,
-                primary_keys=params.primary_keys,
-                sort_keys=params.sort_keys,
-                merge_task_index=index,
-                hash_bucket_count=params.hash_bucket_count,
-                drop_duplicates=params.drop_duplicates,
-                hash_group_index=item[0],
-                num_hash_groups=params.hash_group_count,
-                max_records_per_output_file=params.records_per_compacted_file,
-                enable_profiler=params.enable_profiler,
-                metrics_config=params.metrics_config,
-                s3_table_writer_kwargs=params.s3_table_writer_kwargs,
-                read_kwargs_provider=params.read_kwargs_provider,
-                round_completion_info=round_completion_info,
-                object_store=params.object_store,
-                deltacat_storage=params.deltacat_storage,
-                deltacat_storage_kwargs=params.deltacat_storage_kwargs,
-            )
-        }
+    total_input_records_count = np.int64(0)
+    total_hb_record_count = np.int64(0)
+    telemetry_time_hb = 0
+    if params.hash_bucket_count == 1:
+        local_dfe_list, input_records_count = get_local_delta_file_envelopes(
```

Comment:
Specifically, you are talking about reading the delta within the driver vs. reading the delta within a single merge task? I've since updated the input sizes with more accurate numbers, so we're getting improvements up until ~6.5 GB.